src/java/org/apache/cassandra/net/MessagingService.java

```diff
@@ -527,11 +530,17 @@ public Future<Void> maybeReconnectWithNewIp(InetAddressAndPort address, InetAddr
      */
     public void shutdown()
     {
-        shutdown(1L, MINUTES, true, true);
+        if (NON_GRACEFUL_SHUTDOWN)
+            // this branch is used in unit-tests when we really never restart a node and shutting down means the end of test
+            shutdown(1L, MINUTES, false, true, false);
+        else
+            shutdown(1L, MINUTES, true, true, true);
     }
 
-    public void shutdown(long timeout, TimeUnit units, boolean shutdownGracefully, boolean shutdownExecutors)
+    public void shutdown(long timeout, TimeUnit units, boolean shutdownGracefully, boolean shutdownExecutors, boolean blocking)
```

Comment:
I think this would be cleaner to break into 2 methods, `shutdownGracefully()` and `shutdownAbruptly()` or something, rather than going the multi-bool precondition most of the method is dead on various branches approach we have here.

Reply:
Yes, I agree